sovrin_node/server/node.py

```diff
@@ -86,13 +87,28 @@ def __init__(self,
         self.on_new_ledger_added(CONFIG_LEDGER_ID)
         self.states[CONFIG_LEDGER_ID] = self.loadConfigState()
         self.upgrader = self.getUpgrader()
+        self.poolCfg = self.getPoolConfig()
         self.configReqHandler = self.getConfigReqHandler()
         self.initConfigState()
         self.requestExecuter[CONFIG_LEDGER_ID] = self.executeConfigTxns
 
         self.nodeMsgRouter.routes[Request] = self.processNodeRequest
         self.nodeAuthNr = self.defaultNodeAuthNr()
 
+        self.nodeMsgRouter.routes[PrePrepare] = self.extraPrePrepareProcesing
+
+
+    def extraPrePrepareProcesing(self, msg, frm):
```

Comment:
Please add a test to check that PrePrepares are not processed during read-only state unless this is a CONFIG_LEDGER.
Do we process POOL_UPGRADE when read-only? Please add a test